### ISSUE TITLE
lupdate "Qualifying with unknown namespace/class"

### ISSUE THREAD

**Issue by ghost** [qTox/qTox#1861]
Started happening with commits from about 3-4 days ago.

`lupdate -pro qtox.pro -ts translations/es.ts` outputs

```
/home/path/qTox/src/widget/form/settings/advancedform.cpp:39: Qualifying with unknown namespace/class ::AdvancedForm
/home/path/qTox/src/widget/form/settings/avform.cpp:142: Qualifying with unknown namespace/class ::AVForm
/home/path/qTox/src/widget/form/settings/generalform.cpp:98: Qualifying with unknown namespace/class ::GeneralForm
/home/path/qTox/src/widget/form/settings/generalform.cpp:305: Qualifying with unknown namespace/class ::GeneralForm
/home/path/qTox/src/widget/form/settings/generalform.cpp:361: Qualifying with unknown namespace/class ::GeneralForm
/home/path/qTox/src/widget/friendwidget.cpp:60: Qualifying with unknown namespace/class ::FriendWidget
/home/path/qTox/src/widget/friendwidget.cpp:175: Qualifying with unknown namespace/class ::FriendWidget
/home/path/qTox/src/widget/friendwidget.cpp:266: Qualifying with unknown namespace/class ::FriendWidget
/home/path/qTox/src/widget/tool/chattextedit.cpp:70: Qualifying with unknown namespace/class ::ChatTextEdit
/home/path/qTox/src/widget/widget.cpp:216: Qualifying with unknown namespace/class ::Widget
/home/path/qTox/src/widget/widget.cpp:377: Qualifying with unknown namespace/class ::Widget
/home/path/qTox/src/widget/widget.cpp:388: Qualifying with unknown namespace/class ::Widget
/home/path/qTox/src/widget/widget.cpp:419: Qualifying with unknown namespace/class ::Widget
/home/path/qTox/src/widget/widget.cpp:433: Qualifying with unknown namespace/class ::Widget
/home/path/qTox/src/widget/widget.cpp:444: Qualifying with unknown namespace/class ::Widget
/home/path/qTox/src/widget/widget.cpp:512: Qualifying with unknown namespace/class ::Widget
/home/path/qTox/src/widget/widget.cpp:521: Qualifying with unknown namespace/class ::Widget
/home/path/qTox/src/widget/widget.cpp:629: Qualifying with unknown namespace/class ::Widget
/home/path/qTox/src/widget/widget.cpp:671: Qualifying with unknown namespace/class ::Widget
/home/path/qTox/src/widget/widget.cpp:837: Qualifying with unknown namespace/class ::Widget
/home/path/qTox/src/widget/widget.cpp:895: Qualifying with unknown namespace/class ::Widget
/home/path/qTox/src/widget/widget.cpp:957: Qualifying with unknown namespace/class ::Widget
/home/path/qTox/src/widget/widget.cpp:985: Qualifying with unknown namespace/class ::Widget
/home/path/qTox/src/widget/widget.cpp:1138: Qualifying with unknown namespace/class ::Widget
/home/path/qTox/src/widget/widget.cpp:1218: Qualifying with unknown namespace/class ::Widget
/home/path/qTox/src/widget/widget.cpp:1472: Qualifying with unknown namespace/class ::Widget
```

**Comment by tux3** [comment]
It seems that lupdate does not understand the "final" specifier.

https://bugreports.qt.io/browse/QTBUG-42735

**Comment by tux3** [comment]
This has been fixed upstream in the upcoming Qt 5.6.
